### PR TITLE
Update trait list and type list

### PR DESCRIPTION
**Pull request by bombless**
Update trait list according to [prelude](https://github.com/rust-lang/rust/blob/master/src/libstd/prelude/v1.rs) and [auto deriving](https://github.com/rust-lang/rust/blob/master/src/libsyntax/ext/deriving/mod.rs).

### CODE REVIEW

File: rust.xml [hunk 1]
```diff
@@ -64,29 +64,45 @@
...
+		<item> Show </item>
+		<item> Sized </item>
+		<item> SliceConcatExt </item>
+		<item> SliceExt </item>
+		<item> Str </item>
+		<item> StrExt </item>
+		<item> String </item>
+		<item> Sync </item>
+		<item> ToString </item>
+		<item> Vec </item>
```
Comment (eddyb) [inline rust.xml]: There's a lot of types here that aren't traits.

**Comment by bombless** [comment]
Oh, some spaces were converted to tabs. But most lines of this file start with tabs anyway.

**Comment by eddyb** [comment]
Please use something like `git gui` to remove the irrelevant parts of the diff (e.g. switch to "amend last commit" mode, select the file, select the range, right click and "unstage lines from commit").

File: rust.xml [hunk 2]
```diff
@@ -64,29 +64,42 @@
...
-		<item> Shl </item>
-		<item> Shr </item>
-		<item> Index </item>
-		<item> Not </item>
+		<item> PartialEq </item>
+		<item> PartialOrd </item>
+		<item> PtrExt </item>
+		<item> Rand </item>
+		<item> Send </item>
+		<item> Show </item>
```
Comment (eddyb) [inline rust.xml]: This is deprecated, could we not have it?

**Comment by eddyb** [comment]
I'm not sure I agree with hardcoding a list of `std` traits, but I am not going to get rid of it just yet.
I think anything smarter than a syntax highlighter would ignore these lists - kdevelop does, anyways, in languages it supports.

**Comment by bombless** [comment]
Yeah, I believe an editor don't need a list for known type to do highlighting, especially for a language like Rust which don't have a problem with mixing type names and variable names.
Since now we have a formal grammar recently, maybe this highlighting will be totally rewritten some day this year.

**Comment by bombless** [comment]
And there's [new keyword `macro`](https://github.com/rust-lang/rust/blob/master/src/libsyntax/parse/token.rs#L577) should be added to the keyword list, maybe later.

**Comment by eddyb** [comment]
Thanks!

**Pull request merged**